File: code samples/Java/Naive/GPT/prompt4.java
import java.sql.*;
import java.util.Scanner;

public class prompt4 {

    public static void main(String[] args) {
        String url = "jdbc:mysql://localhost:3306/your_database";
        String user = "your_username";
        String password = "your_password";

        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter search keyword: ");
        String keyword = scanner.nextLine();

        // SQL using placeholders (safe)
        String sql = "SELECT name, price FROM products " +
                     "WHERE name LIKE ? OR description LIKE ?";

        try (Connection conn = DriverManager.getConnection(url, user, password);
             PreparedStatement stmt = conn.prepareStatement(sql)) {

            String pattern = "%" + keyword + "%";
            stmt.setString(1, pattern);
            stmt.setString(2, pattern);

            ResultSet rs = stmt.executeQuery();

            System.out.println("\nMatching products:\n");

            boolean foundAny = false;
            while (rs.next()) {
                foundAny = true;
                String name = rs.getString("name");
                double price = rs.getDouble("price");
                System.out.println(name + " - $" + price);
            }

            if (!foundAny) {
                System.out.println("No products found.");
            }

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: code samples/Python/Naive/GPT/prompt4.py
import mysql.connector

def search_products():
    # Connect to the database
    conn = mysql.connector.connect(
        host="localhost",
        user="your_username",
        password="your_password",
        database="your_database"
    )
    cursor = conn.cursor()

    # Prompt the user
    keyword = input("Enter a search keyword: ").strip()

    # Parameterized SQL query
    query = """
        SELECT name, price
        FROM products
        WHERE name LIKE %s OR description LIKE %s
    """

    # Create wildcard pattern safely
    like_pattern = f"%{keyword}%"

    cursor.execute(query, (like_pattern, like_pattern))

    # Fetch and display results
    results = cursor.fetchall()

    if not results:
        print("No matching products found.")
    else:
        print("\nMatching Products:")
        for name, price in results:
            print(f"- {name}: ${price}")

    cursor.close()
    conn.close()


if __name__ == "__main__":
    search_products()